src/url-loader/package-url-resolver.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import {posix as pathlib} from 'path';
+import {parse as parseUrl, Url} from 'url';
+import {UrlResolver} from './url-resolver';
+
+export interface PackageUrlResolverOptions {
+  componentDir?: string;
+  hostname?: string
+}
+
+/**
+ * Resolves a URL to a canonical URL within a package.
+ */
+export class PackageUrlResolver implements UrlResolver {
+
+  componentDir: string;
+  hostname: string;
+
+  constructor(options?: PackageUrlResolverOptions) {
+    options = options || {};
+    this.componentDir = options.componentDir || 'bower_components/';
+    this.hostname = options.hostname || null;
+  }
+
+  canResolve(url: string): boolean {
```

Comment:
Document what resolution already needs to have taken place on this url. Also we want tests.